### DISCUSSION TITLE
Can we please remove tight checking for `${number}` at parseUnits

### DISCUSSION THREAD

**Discussion by vseehausen** [wevm/viem#490] [Idea / Feature Request]
Hey 

parseUnits() expects a "`${number}`" type, while even formatUnits does return a string.

This overcomplicates type checking while adding pretty little value, so I would highly vouch for accepting strings in this cases.

This also counts for Address / `0x${string}` at abitypes, but that is acceptable with the conversion functions.

**Comment by kdembler** [comment]
Yes please! This is our codebase:
```ts
import { parseUnits as viemParseUnits } from 'viem';
// workaround for weird viem types
export function parseUnits(value: string, decimals: number) {
  return viemParseUnits(value as `${number}`, decimals);
}
```

Example of code that gave us issues:
```ts
import { parseUnits } from 'viem';
const num = 12
parseUnits(num.toString(), 10)
```

**Reply by jxom** [reply]
Yeah, maybe this type is too strict. Can look into laxxing it

**Reply by MantisClone** [reply]
Here's how I solved for this.

```typescript
import { parseUnits } from 'viem';
const num = 12
parseUnits(`${num}`, 10)
```

EDIT: Don't use this. Storing as `number` leads to loss in precision. Better to store as `string` and cast to `${number}`.

**Comment by mikemcdonald** [comment]
Another example of unexpected errors due to the `${number}` checking:

```
const test: `${number}` = '10100000000000000020000';
parseUnits(test, 0);
```

`SyntaxError: Cannot convert 1.0100000000000001e+22 to a BigInt`

This is a real live example of a token with 0 decimals and the balance shown that fails

**Comment by llllvvuu** [comment]
Here's a no-runtime-overhead workaround for this:
```ts
// ./src/overrides.d.ts
declare global {
  module "viem" {
    export * from "viem"
    export function parseUnits(value: string, decimals: number): bigint
  }
}
```

**Comment by DannyDelott** [comment]
Bump on this @jxom, happy to open a PR to widen the type.
